build-tools/src/main/kotlin/org/jetbrains/kotlin/MPPTools.kt

```diff
@@ -115,7 +116,16 @@ fun mergeReports(reports: List<File>): String {
         val reportElement = JsonTreeParser.parse(json)
         BenchmarksReport.create(reportElement)
     }
-    return if (reportsToMerge.isEmpty()) "" else reportsToMerge.reduce { result, it -> result + it }.toJson()
+    val structuredReports = mutableMapOf<String, MutableList<BenchmarksReport>>()
+    reportsToMerge.map { it.compiler.backend.flags.joinToString() to it }.forEach {
+        if (it.first in structuredReports) {
```

Comment:
It's possible to use [`getOrPut`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/get-or-put.html) here:
```kotlin
structuredReports.getOrPut(it.first) { mutableListOf<BenchmarksReport>() }.add(it.second)
```